iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -1119,6 +1131,365 @@ rapidjson::Document IModelJsECPresentationSerializer::_AsJson(ContextR ctx, Node
     return json;
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static Utf8String GetECValueTypeName(PrimitiveType type)
+    {
+    switch (type)
+        {
+        case PRIMITIVETYPE_Binary:
+            return "binary";
+        case PRIMITIVETYPE_Boolean:
+            return "boolean";
+        case PRIMITIVETYPE_DateTime:
+            return "dateTime";
+        case PRIMITIVETYPE_Double:
+            return "double";
+        case PRIMITIVETYPE_Integer:
+            return "int";
+        case PRIMITIVETYPE_Long:
+            return "long";
+        case PRIMITIVETYPE_Point2d:
+            return "point2d";
+        case PRIMITIVETYPE_Point3d:
+            return "point3d";
+        case PRIMITIVETYPE_String:
+            return "string";
+        default:
+            return "";
+        }
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static int ParsePrimitiveType(Utf8String const& typeName)

```

Comment:
- this function could take `Utf8CP` - at the moment callers get it from json and have to copy it into `Utf8String` just to call this function

- please don't do
  ```cpp
  if (x)
    return 1;
  else if (y)
    return 2;
  else
    return 3;
  ```
  instead, do this:
  ```cpp
  if (x)
    return 1;
  if (y)
    return 2;
  return 3;
  ```
- please make sure all callers check return value for `0`

Reply:
Changed to Utf8CP.
Removed the unnecessary `else`s.
Added checks for `0`.